### ISSUE TITLE
Back button handling in in MainTabs

### ISSUE THREAD

**Issue by ShashwatMDas** [zulip/zulip-mobile#4353]
Current Behaviour: On pressing the android/IOS back button in any of the main tabs closes the application.

Ideal Behaviour: Pressing the back button should take the user back to the Home Tab. 

Alternative Behaviour: Should show a toast/snackbar, asking the user to press the back button twice to exit the app.

**Comment by chrisbobbe** [comment]
> Current Behaviour: On pressing the android/IOS back button in any of the main tabs closes the application.

This sounds like an Android-only thing; have you tested on iOS? 🙂

**Comment by ShashwatMDas** [comment]
No, I have tried only for android. Sorry for the misinformation, will edit it.

**Comment by chrisbobbe** [comment]
> Ideal Behaviour: Pressing the back button should take the user back to the Home Tab.

This sounds fine to me; you mentioned in [discussion](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Back.20button.20handling.20in.20in.20MainTabs/near/1085678) on chat.zulip.org that this is what Slack does on Android. (I'm more familiar with iOS than Android; Greg may have other ideas.) 🙂 

First, I'd recommend looking at [`createMaterialBottomTabNavigator`'s `backBehavior` option](https://reactnavigation.org/docs/4.x/material-bottom-tab-navigator/). Would setting that to `initialRoute`, and setting `initialRouteName` to `'home'`, in `MainTabs`, do the job? 🙂 You could also take a look at the other values `backBehavior` could take—e.g., `history` would take the user back through each tab they'd previously visited, instead of jumping straight to the home tab.

If that doesn't work (give it a try first, and see if you can find out what might be going wrong! 🙂), we may have to look more closely at our use of React Native's `BackHandler` for Android, in src/nav/BackNavigationHandler.js, and our custom "go-back" action creator, `navigateBack`, in src/nav/navActions.js.

**Comment by ShashwatMDas** [comment]
@chrisbobbe can your review this [PR](https://github.com/zulip/zulip-mobile/pull/4354) for this issue?

**Comment by Aman-Maheshwari** [comment]
@chrisbobbe  Is this issue still active ? can I work on it ?

**Comment by Aman-Maheshwari** [comment]
@chrisbobbe Please review this PR. Thank you. :relaxed:

**Comment by chrisbobbe** [comment]
Hi, @Aman-Maheshwari! @gnprice has left a comment about this issue on CZO, [here](https://chat.zulip.org/#narrow/stream/48-mobile/topic/Back.20button.20handling.20in.20in.20MainTabs/near/1086157—let's continue the conversation there. 🙂

I've also noticed that #2716 contains more conversation about this issue; I'll make a note about that at that in that CZO thread.
